### PR TITLE
[KYUUBI #3208] Fix Flaky Test - MetadataManagerSuite: metadata request metrics

### LINKED ISSUE #3208: Flaky Test: MetadataManagerSuite - metadata request metrics
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the flaky test

MetadataManagerSuite:
```
- metadata request metrics *** FAILED ***
  2 did not equal 1 (MetadataManagerSuite.scala:127)
```

### Actions URL

https://github.com/apache/incubator-kyuubi/runs/7738368898?check_suite_focus=true

### Parent issue

#1733

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Close #3208 
Fix flaky test:
```
MetadataManagerSuite:
- metadata request metrics *** FAILED ***
  2 did not equal 1 (MetadataManagerSuite.scala:141)
```

Cleanup the retrying metadata to prevent flaky test.
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/test/scala/org/apache/kyuubi/server/metadata/MetadataManagerSuite.scala [hunk 1]
```diff
@@ -96,6 +96,9 @@ class MetadataManagerSuite extends KyuubiFunSuite {
       assert(!retryRef2.hasRemainingRequests())
       assert(metadataManager.getBatch(metadata2.identifier).getState === "RUNNING")
     }
+    metadataManager.deRegisterRequestsRetryRef(metadata.identifier)
+    metadataManager.deRegisterRequestsRetryRef(metadata2.identifier)
+    Thread.sleep(conf.get(KyuubiConf.METADATA_REQUEST_RETRY_INTERVAL) * 2)
```
[the remark this answers is not in the document]
Comment (turboFei) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/server/metadata/MetadataManagerSuite.scala:101]: Will check it

[the remark this answers is not in the document]
Comment (turboFei) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/server/metadata/MetadataManagerSuite.scala]: updated

**Comment by turboFei** [comment]
cc @pan3793

**Comment by turboFei** [comment]
cc @pan3793

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master and branch-1.6

**Pull request closed without merging**